### DISCUSSION TITLE
How to add record to onetomany field using Dao in Golang

### DISCUSSION THREAD

**Discussion by hel8er** [pocketbase/pocketbase#2835] [Q&A] [answered]
Hi everyone! Need Help. I cant set field with multiple relation (collection keys has relation field users to collection tg_users. Maybe its wrong way, but i trying use MergeExpand method.
I guess, my expand struct is wrong

```golang
	key_record, _ := app.PB.Dao().FindRecordById("keys", "frs8q9laft6hbsp")
	user, _ := app.PB.Dao().FindRecordById("tg_users", <id>)
	var expand map[string]any
	expand = make(map[string]any)
	expand["users"] = user // maybe wrong
	key_record.MergeExpand(expand)
```

**Comment by ganigeorgiev** [comment, marked as answer]
The `MergeExpand` method is to merge the existing "expand" record field with the provided map (eg. when [expanding relations](https://pocketbase.io/docs/expanding-relations/) is used).

If you want only to set a record field value you can use the `Set` method, eg:

```go
key, err := app.PB.Dao().FindRecordById("keys", "frs8q9laft6hbsp")
if err != nil {
    return err
}

// get the existing user ids
userIds := key.GetStringSlice("users")

// append the new id
userIds = append(userIds, "YOUR_NEW_USER_ID")

// update the key.users value
key.Set("users", userIds)

if err := app.PB.Dao().SaveRecord(key); err != nil {
    return err
}
```
(_you can also check https://pocketbase.io/docs/record-methods/#getset-record-fields; more detailed documentation will be available with the next release_)

**Reply by hel8er** [reply]
Its resolved my problem, thank you!
